src/extension/NUnit2FrameworkDriver.cs

```diff
@@ -66,16 +64,37 @@ public NUnit2FrameworkDriver(AppDomain testDomain)
 
             var initializer = DomainInitializer.CreateInstance(_testDomain);
             initializer.InitializeDomain((int)InternalTrace.Level);
-
-            _runner = RemoteTestRunner.CreateInstance(_testDomain, 1);
         }
 
         public string ID { get; set; }
 
+        private TestRunner Runner {
+            get
+            {
+                if (_runner == null)
+                {
+                    int runnerId;
+                    if (ID == null)
+                    {
+                        throw new NUnitEngineException("NUnit 2 Driver requires that ID must be defined at first.");
+                    }
```

Comment:
@CharliePoole we already discussed this exception in my previous PR. And I thought that InvalidOperationException is preferable here, but after discussion I understood that NUnitEngineException it is more common way for NUnit (and also is more informative from my point of view)